crates/wdl-engine/src/backend/lsf_apptainer.rs

```diff
@@ -0,0 +1,656 @@
+const APPTAINER_COMMAND_FILE_NAME: &str = "apptainer_command";
+
+/// The root guest path for inputs.
+const GUEST_INPUTS_DIR: &str = "/mnt/task/inputs/";
+
+/// The guest working directory.
+const GUEST_WORK_DIR: &str = "/mnt/task/work";
+
+/// The guest path for the command file.
+const GUEST_COMMAND_PATH: &str = "/mnt/task/command";
+
+/// The path to the container's stdout.
+const GUEST_STDOUT_PATH: &str = "/mnt/task/stdout";
+
+/// The path to the container's stderr.
+const GUEST_STDERR_PATH: &str = "/mnt/task/stderr";
+
+/// The maximum length of an LSF job name.
+///
+/// See <https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=o-j>.
+const LSF_JOB_NAME_MAX_LENGTH: usize = 4094;
+
+#[derive(Debug)]
+struct LsfApptainerTaskRequest {
+    backend_config: Arc<LsfApptainerBackendConfig>,
+    name: String,
+    spawn_request: TaskSpawnRequest,
+    /// The requested container for the task.
+    container: String,
+    /// The requested CPU reservation for the task.
+    cpu: f64,
+    /// The requested memory reservation for the task, in bytes.
+    memory: u64,
+    /// The broadcast channel to update interested parties with the status of
+    /// executing tasks.
+    ///
+    /// This backend does not yet take advantage of the full Crankshaft
+    /// machinery, but we send rudimentary messages on this channel which helps
+    /// with UI presentation.
+    crankshaft_events: Option<broadcast::Sender<Event>>,
+    cancellation_token: CancellationToken,
+}
+
+impl TaskManagerRequest for LsfApptainerTaskRequest {
+    fn cpu(&self) -> f64 {
+        self.cpu
+    }
+
+    fn memory(&self) -> u64 {
+        self.memory
+    }
+
+    async fn run(self) -> anyhow::Result<super::TaskExecutionResult> {
+        let crankshaft_task_id = crankshaft::events::next_task_id();
+
+        let container_sif = sif_for_container(
+            &self.backend_config,
+            &self.container,
+            self.cancellation_token.clone(),
+        )
+        .await?;
+
+        let attempt_dir = self.spawn_request.attempt_dir();
+
+        // Create the host directory that will be mapped to the WDL working directory.
+        let wdl_work_dir = attempt_dir.join(WORK_DIR_NAME);
+        fs::create_dir_all(&wdl_work_dir).await.with_context(|| {
+            format!(
+                "failed to create WDL working directory `{path}`",
+                path = wdl_work_dir.display()
+            )
+        })?;
+
+        // Write the evaluated WDL command section to a host file.
+        let wdl_command_path = attempt_dir.join(COMMAND_FILE_NAME);
+        fs::write(&wdl_command_path, self.spawn_request.command())
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to write WDL command contents to `{path}`",
+                    path = wdl_command_path.display()
+                )
+            })?;
+        #[cfg(unix)]
+        fs::set_permissions(&wdl_command_path, Permissions::from_mode(0o777)).await?;
```

Comment:
I wonder if we'd want to do like `0o770` or, alternatively, make this configurable from the start?